packages/firestore/src/platform_browser/webchannel_connection.ts

```diff
@@ -123,12 +123,24 @@ export class WebChannelConnection implements Connection {
                 xhr.getResponseText()
               );
               if (status > 0) {
-                reject(
-                  new FirestoreError(
-                    mapCodeFromHttpStatus(status),
-                    'Server responded with status ' + xhr.getStatusText()
-                  )
-                );
+                const response = xhr.getResponseJson().error;
+                if (!!response) {
+                  const serverError = response.status
+                    .toLowerCase()
+                    .replace('_', '-');
+                  const firestoreError: Code =
```

Comment:
Since this is a `Code` not a `FirestoreError` I'd rename to avoid confusion (`firestoreErrorCode` or whatever would be fine).